### ISSUE TITLE
[HTML] Add isPrimary boolean property to the Action class

### ISSUE THREAD

**Issue by dptejas** [Microsoft/AdaptiveCards#1515]
# Problem
For backwards compatibility with its previous Actionable Message card format, Outlook needs the ability to mark an action as "primary" so that it can get a different visual treatment.

# Solution
In the HTML renderer:
- Introduce an **isPrimary** boolean property on the Action class
- Introduce a way to custom parse Action objects (similarly to the custom parsing of Element objects)

# Design
New isPrimary property on the Action class:
```typescript
export class Action {
    ...
    isPrimary: boolean = false;
    ...
}
```

Custom action parsing events (global and per-card, like other events):
```typescript
export class AdaptiveCard {
    ...
    static onParseAction: (action: Action, json: any) => void = null;
    onParseAction: (action: Action, json: any) => void = null;
    ...
}
```

It is the responsibility of the host to handle the onParseAction event to custom parse the isPrimary action. isPrimary is not parsed out of the box and is not an official Adaptive property.

**Comment by ignacionr** [comment]
Hello @dptejas 
The property you mention does not exist on the current http://adaptivecards.io/schemas/adaptive-card.json ; can you share where you've found it documented?

**Comment by dclaux** [comment]
@ignacionr the property is indeed not documented. It will be introduced in the HTML renderer only, not in the schema (at least not for now.) Hosts who want to support this property will have to do so via extensibility (e.g. by custom parsing it.) THis property will make it possible to give a different visual treatment to the "primary".


I have updated the title and will add a slittle spec.

**Comment by ignacionr** [comment]
This makes sense. Can you guys make sure that when rendered on HTML, pressing “enter” while the focus is on a control will execute this action?
It could be convenient.

**Comment by dclaux** [comment]
Yes, we can do that. I just opened https://github.com/Microsoft/AdaptiveCards/issues/1517

**Comment by matthidinger** [comment]
For reference, we have a few related issues on this topic: #861 and #272

**Comment by dclaux** [comment]
https://github.com/Microsoft/AdaptiveCards/issues/861 is obsolete IMO, I no longer believe the solution I proposed in there is the right one. I think we should close it.

**Comment by dptejas** [comment]
Implemented solution in #1521
